tests/serverpod_test_server/test_integration/websockets/method_websockets/method_streams/revoked_authentication_stream_test.dart

```diff
@@ -0,0 +1,255 @@
+import 'package:serverpod_test_server/test_util/config.dart';
+import 'package:serverpod_test_server/test_util/test_key_manager.dart';
+import 'package:serverpod_test_server/test_util/test_serverpod.dart';
+import 'package:test/test.dart';
+
+void main() {
+  late Serverpod server;
+  late Session session;
+  late c.Client client;
+  late TestAuthKeyManager authKeyManager;
+  var authenticatedUserId = 1;
+  var tokenCounter = 0;
+
+  setUp(() async {
+    // This creates a server with an authentication handler that always returns
+    // a user with Admin scope.
+    var tokenCounter = 0;
+    server = IntegrationTestServer.create(
+      authenticationHandler: (session, token) async => AuthenticationInfo(
+        authenticatedUserId,
+        {Scope.admin},
+        authId: 'token-${tokenCounter++}',
+      ),
+    );
+    await server.start();
+    session = await server.createSession();
+
+    authKeyManager = TestAuthKeyManager();
+    client = c.Client(
+      serverUrl,
+      authenticationKeyManager: authKeyManager,
+    );
+  });
+
+  tearDown(() async {
+    await session.close();
+    client.closeStreamingMethodConnections(exception: null);
+    await server.shutdown(exitProcess: false);
+    client.close();
+  });
+
+  group('Given an authenticated user', () {
+    late String authToken;
+    setUp(() async {
+      authToken = const Uuid().v4();
+      authKeyManager.put(authToken);
+    });
+    group('connected to an authenticated streaming method', () {
+      late Completer<dynamic> streamClosedCompleter;
+      late Completer<int> valueReceivedCompleter;
+      late StreamController<int> inStream;
+
+      setUp(() async {
+        streamClosedCompleter = Completer<dynamic>();
+        inStream = StreamController<int>();
+        Stream<int> outStream;
+        {
+          outStream = client.authenticatedMethodStreaming
+              .intEchoStream(inStream.stream);
+          valueReceivedCompleter = Completer<int>();
+          outStream.listen((event) {
+            if (valueReceivedCompleter.isCompleted) {
+              return;
+            }
+            valueReceivedCompleter.complete(event);
+          }, onError: (e) {
+            streamClosedCompleter.complete(e);
+          });
+
+          inStream.add(1);
+          // Validate that the stream works
+          await valueReceivedCompleter.future;
+          assert(valueReceivedCompleter.isCompleted);
+        }
+      });
+
+      test(
+          'when the authenticated user is revoked then the stream is closed with error.',
+          () async {
+        await expectLater(
+          session.messages.revokedAuthentication(
+            authenticatedUserId,
+            RevokedAuthenticationUser(),
+          ),
+          completion(true),
+        );
+
+        await expectLater(streamClosedCompleter.future, completes);
+        var exception = await streamClosedCompleter.future;
+        expect(exception, isA<ServerpodClientException>());
```

Comment:
Question: what data are we sending with this exception? Would perhaps be nice to validate that too, just to ensure we are not leaking anything.

Reply:
I think we should probably close using another exception. Will change it to tthat instead. This is not a regulair client exception.